src/activities/quizzes/attempts/QuizAttemptsEntity.js

```diff
@@ -106,6 +128,33 @@ export class QuizAttemptsEntity extends Entity {
 		return this._entity && this._entity.getSubEntityByClass(Classes.quizzes.attempts.retakeIncorrectOnly);
 	}
 
+	/**
+	 * @returns {object} quiz attempt conditions sub-entity
+	 */
+	getAttemptConditionsSubEntity() {
+		if (this._entity.hasSubEntityByClass(Classes.quizzes.attempts.attemptConditions) && this._entity.hasSubEntityByClass('collection')) {
+			return this._entity && this._entity.getSubEntityByClass(Classes.quizzes.attempts.attemptConditions);
+		}
+
+		return null;
+	}
+
+	/**
+	 * @returns {object} a singular quiz attempt condition sub-entity
+	 */
+	getAttemptConditionSubEntity(attemptConditionNumber) {
+		const entity = this.getAttemptConditionsSubEntity();
+		const attemptConditionEntities = entity.getSubEntitiesByClass(Classes.quizzes.attempts.attemptCondition);
+		if (!attemptConditionEntities) return;
+		const attemptConditionEntity = attemptConditionEntities.find((entity) => {
+			if (!entity.properties || !entity.properties.attempt) return false;
+			return entity.properties.attempt === attemptConditionNumber ? entity : false;

```

Comment:
this can just be `return entity.properties.attempt === attemptConditionNumber;` I believe

Reply:
Yes, thanks for catching that nonsense.